app/controllers/brand_selection_controller.rb

```diff
@@ -0,0 +1,15 @@
+class BrandSelectionController < ApplicationController
+  before_action :authenticate_user!
+
+  def show
+    # If user has a current brand, redirect to it
+    if current_user.current_brand
+      locale = params[:locale] || I18n.locale || I18n.default_locale
+      redirect_to "/#{current_user.current_brand.slug}/#{locale}"
```

Comment:
this should look like `redirect_to edit_brand_path(current_user.current_brand.slug, locale)`